pkg/targetgroup/targetgroup.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package targetgroup
+
```

Comment:
It's a shared concept between SD and retrieval, so it was kind of unclear where to put it. A shared place that isn't tied to either seemed to make sense.

Reply:
I feel SD is where is really belongs. It's the output of SD.

Reply:
`discovery` or `discovery/config` then? The latter would make it more decoupled, but `config` is only part of what target groups are used for (the output of SDs and input to retrieval being the other use).

Reply:
I'd say `discovery`. We should probably also split out the two different uses of this struct (SD output vs. static_config input) into separate structs as the struct doesn't match the JSON. Probably not for this PR though.

Reply:
Good point, agreed.

Reply:
This can't go into `manager.go` in the `discovery` package because it'll create an import cycle.

Reply:
This can't go in either of those because it'll create import cycles.
For `discovery`, config imports `manager.go` already imports `config`
For `discovery/config`, it creates import cycles with all other SD discovery packages as `discovery/config` imports all of them.